Userland/Libraries/LibGfx/ImageFormats/ILBMLoader.cpp

```diff
@@ -129,9 +133,30 @@ static ErrorOr<RefPtr<Gfx::Bitmap>> chunky_to_bitmap(ILBMLoadingContext& context
     dbgln_if(ILBM_DEBUG, "created Bitmap {}x{}", width, height);
 
     for (int row = 0; row < height; ++row) {
+        // Keep color: in HAM mode, current color
+        // may be based on previous color instead of coming from
+        // the palette.
+        Color color = Color::Black;
         for (int col = 0; col < width; ++col) {
             u8 index = chunky[(width * row) + col];
-            bitmap->set_pixel(col, row, context.color_table[index]);
+            if (index < context.color_table.size()) {
+                color = context.color_table[index];
+            } else {
```

Comment:
Does entering this block imply that `has_flag(context.viewport_mode, ViewportMode::HAM)` is true? Could we add a check for this?

Reply:
Yes it does. I'll add the check. What should we do if the flag is not set (malformed files?): throw an error?

Reply:
Yes, returning an error would be best in this case. As you say, the error would only be triggered by a malformed file.

Reply:
I added a check.